### PR TITLE
Add modules specific for FreeBSD.  Keep modules specific for Linux.

### PR DESCRIPTION
**Pull request by pkgdemon**

### CODE REVIEW

**Comment by williamh** [comment]
Why add a separate modules service for FreeBSD instead of making the current  modules service able to  run on both FreeBSD and Linux?

File: conf.d/Makefile [hunk 1]
```diff
@@ -22,6 +22,9 @@ CONF-NetBSD=	moused rarpd savecore
 
 include ${MK}/scripts.mk
 
+modules: modules${SFX}
+	cp $@${SFX} $@
```
Comment (dwfreed) [inline conf.d/Makefile]: Eww... See the next target for a much better way to do this (we really should find a way to make these out of generic targets, though)

File: init.d/modules.BSD.in [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+start()
+{
+	local cnt=0 x
+	for x in $modules
+	do
+		ebegin "Loading module $x"
+		kldload "$x";
+		eend $? "Failed to load $x" && : $(( cnt += 1 ))
+	done
+	if [ $cnt -eq 1 ]; then
```
Comment (dwfreed) [inline init.d/modules.BSD.in]: Why bother? There's nothing wrong with "Autoloaded $x module(s)"

File: init.d/modules.BSD.in [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+start()
+{
+	local cnt=0 x
+	for x in $modules
+	do
+		ebegin "Loading module $x"
+		kldload "$x";
+		eend $? "Failed to load $x" && : $(( cnt += 1 ))
```
Comment (dwfreed) [inline init.d/modules.BSD.in]: I realize the Linux modules initscript does this, but it's really rather gross.  I think we can get just rid of the count entirely.

**Comment by williamh** [comment]
I recommend starting with the current etc/init.d/modules, renaming start() to Linux_modules(), adding FreeBSD_modules() which would be the start function in your new service, then having start() call the appropriate function based on $RC_UNAME.

**Comment by williamh** [comment]
Also, I'm fine with the modules target in conf.d/Makefile at this point. We do similar things with network and staticroute.

**Comment by dwfreed** [comment]
My issue is with the way he implemented the target, not with the target's existence; compare it with the network target.

**Comment by williamh** [comment]
The other way to do the conf.d modules file could be to add comments explaining that on FreeBSD the only variable that works is modules, that way we would just have one conf.d/modules file. But, would that end up being confusing for users?

**Comment by williamh** [comment]
The network target combines two files -- network.in and network.{sfx} to create network.
A better comparison is the staticroute target.

**Comment by williamh** [comment]
After chatting on irc, I understand what @dwfreed is talking about.
The rule in the modules target should be:
cp $^ $@

File: init.d/modules.BSD.in (hunk 3 above)
Comment (pkgdemon) [inline init.d/modules.BSD.in]: Are you referring to removing both lines 12, 17, and replacing 17 with "Autoloaded $x module(s)"?  FreeBSD uses kldload.  I did not quite understand how Linux was loading the module from the Linux example.

Comment (dwfreed) [inline init.d/modules.BSD.in]: Just completely remove counting, and get rid of the "Autoloaded $x module(s)" message entirely.  I made the comment below before I got to this one.

**Comment by williamh** [comment]
@dwfreed, we can talk about removing the counting as a separate change from this. I am quite cautious about it because it will be a user-facing change that I do not see the reason for.

@demonpkg, Thanks for the code for loading modules on FreeBSD. I'm not a FreeBSD user myself, so any help in that area is greatly appreciated. I hope you will continue to contribute. :-)

I used your code for FreeBSD, and came up with an approach that I think fits the design of OpenRC better. Since the service scripts can detect which operating system they are running on by testing against $RC_UNAME, There is no reason to have separate modules service scripts or configuration files for Linux and FreeBSD.

I reworked the current modules service to support them both.

This pull request will close when I commit my change to master.

**Pull request closed without merging**